server/dex/dex.go

```diff
@@ -139,19 +138,26 @@ type configResponse struct {
 	configEnc json.RawMessage
 }
 
-func newConfigResponse(cfg *DexConf, cfgAssets []*msgjson.Asset, cfgMarkets []*msgjson.Market) (*configResponse, error) {
+func newConfigResponse(cfg *DexConf, regAssets map[string]*msgjson.FeeAsset, cfgAssets []*msgjson.Asset, cfgMarkets []*msgjson.Market) (*configResponse, error) {
 	configMsg := &msgjson.ConfigResult{
 		BroadcastTimeout: uint64(cfg.BroadcastTimeout.Milliseconds()),
 		CancelMax:        cfg.CancelThreshold,
-		RegFeeConfirms:   uint16(cfg.RegFeeConfirms),
 		Assets:           cfgAssets,
 		Markets:          cfgMarkets,
-		Fee:              cfg.RegFeeAmount,
 		APIVersion:       uint16(APIVersion),
 		BinSizes:         apidata.BinSizes,
 		DEXPubKey:        cfg.DEXPrivKey.PubKey().SerializeCompressed(),
+		RegFees:          regAssets,
 	}
 
+	// Set the deprecated DCR fee fields.
+	dcrAsset := regAssets["dcr"]
+	if dcrAsset == nil {
+		return nil, fmt.Errorf("DCR is required as a fee asset for backward compatibility")
+	}
```

Comment:
Do this before `configMsg := &msgjson.ConfigResult{`, and you can just set the two fields in the struct literal.